### PR TITLE
Fix cached dir for timm & hugging-face

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

- Change XDG_CACHE_HOME

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by yunchu — DISMISSED** [review]

File: src/otx/__init__.py [hunk 1]
```diff
@@ -5,8 +5,14 @@
 
 __version__ = "2.2.0rc0"
 
+import os
+from pathlib import Path
+
 from otx.core.types import *  # noqa: F403
 
+# Pretrained weight is saved into ~/.cache/torch/hub/checkpoints
+os.environ["XDG_CACHE_HOME"] = str(Path.home() / ".cache" / "torch" / "hub" / "checkpoints")
```
Comment (eunwoosh) [inline src/otx/__init__.py]: do we need to make a directory if it doesn't exist?

Comment (harimkang) [reply src/otx/__init__.py]: When I deleted and restarted it, it created the folder on its own when downloading.

**Review by eunwoosh — DISMISSED** [review]

**Review by sungchul1 — DISMISSED** [review]

File: src/otx/__init__.py [hunk 2]
```diff
@@ -5,8 +5,18 @@
...
+import os
+from pathlib import Path
+
 from otx.core.types import *  # noqa: F403
 
+# Set the value of XDG_CACHE_HOME to set the cache folder that stores the pretrained weights for timm and huggingface.
+# Default, Pretrained weight is saved into ~/.cache/torch/hub/checkpoints
+os.environ["XDG_CACHE_HOME"] = os.getenv(
+    "XDG_CACHE_HOME",
+    str(Path.home() / ".cache" / "torch" / "hub" / "checkpoints"),
```
Comment (yunchu) [inline src/otx/__init__.py:18]: I think torch hub cache path would be like "~/.cache/torch/hub/checkpoints/torch/hub" as the result of this change.
please check here: https://pytorch.org/docs/stable/hub.html#where-are-my-downloaded-models-saved

Comment (harimkang) [reply src/otx/__init__.py:18]: hmm.. let me check..

Comment (harimkang) [reply src/otx/__init__.py:18]: Yes you're right, I'll find another way.

Comment (harimkang) [reply src/otx/__init__.py:18]: https://github.com/openvinotoolkit/training_extensions/pull/3914/commits/24ef76c4175921d50ee2ad0271546e1d64a74d54 - i change `XDG_CACHE_HOME` to `HF_HUB_CACHE`

**Review by sungchul1 — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

**Review by yunchu — APPROVED** [review]

**Pull request merged**